src/dbSta/src/dbNetwork.cc

```diff
@@ -1695,6 +1698,32 @@ void dbNetwork::makeTopCell()
   }
 }
 
+//
+// For fast access we set up hash tables
+// These are not part of the db schema and we build them on the fly
+// after reading in the database.
+//
+
+void dbNetwork::makeAccessHashes()
```

Comment:
Why aren't we doing this in odb as the objects are created rather than after the fact from outside.  It seems much more error prone as anyone adding a new object will have to be aware of hash updating.

Reply:
The unordered hashes are in the low level dbModule as user fields and populated/depopulated using the low level odb create/destroy. So anyone adding/deleting a new object does not need to be aware of the hash updating as it is done in the low level odb routines (see for example: dbModInst::create in odb/src/db/dbModInst.cpp). The observer routine makeAccessHashes is used in the case when the database is loaded from memory which does not explicitly call the low level create routines. The observer call is readDbNetlistAfter. The use of the observer routines seems to be a pattern used in the code to set things up after loading the database.